addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRuleUnitTest.java

```diff
@@ -456,12 +469,39 @@ void shouldRaiseAlertWhenThresholdLowAndMessageOutOfScope() throws URIException
         assertEquals(1, alertsRaised.size());
     }
 
+    private static Stream<Arguments> provideNonLowThresholdsAndBooleans() {
+        return Stream.of(
+                Arguments.of(AlertThreshold.MEDIUM, true),
+                Arguments.of(AlertThreshold.MEDIUM, false),
+                Arguments.of(AlertThreshold.HIGH, true),
+                Arguments.of(AlertThreshold.HIGH, false));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideNonLowThresholdsAndBooleans")
+    void shouldNotRaiseAlertOnGetAtNonLowThresholdRegardlessOfScope(
+            AlertThreshold threshold, boolean isInScope) throws URIException {
+        // Given
+        rule.setCSRFIgnoreAttName("ignore");
+        HttpMessage msg = createScopedMessage(isInScope, true);
+        // When
+        rule.setConfig(new ZapXmlConfiguration());
+        rule.setAlertThreshold(threshold);
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
     void formWithoutAntiCsrfToken() {
         msg.setResponseBody(
                 "<html><head></head><body><form id=\"no_csrf_token\"><input type=\"text\"/><input type=\"submit\"/></form></body></html>");
     }
 
     private HttpMessage createScopedMessage(boolean isInScope) throws URIException {
+        return createScopedMessage(isInScope, false);
+    }
+
+    private HttpMessage createScopedMessage(boolean isInScope, boolean isGet) throws URIException {
```

Comment:
This one could be updated to accept an HTTP method too.

Reply:
Oops, yup